### PR TITLE
Reject empty required fields in the backend

### LINKED ISSUE #50: Backend should reject request with required field empty
For example, item have `name` as required in openapi.yaml, it should not accept post with empty name.

### PR DESCRIPTION
**Pull request by josueetcom**
Additionally added unit tests for potential bad requests like these.

Closes #50

### CODE REVIEW

File: backend/api-service/src/api_inventory_service.go [hunk 1]
```diff
@@ -174,6 +174,15 @@ func (s *InventoryApiService) ListLocations(w http.ResponseWriter) error {
...
 // NewInventoryTransaction - Create a new Inventory Transaction
 func (s *InventoryApiService) NewInventoryTransaction(inventoryTransaction InventoryTransaction, w http.ResponseWriter) error {
+	if inventoryTransaction.Action == "" {
+		return EncodeJSONStatus(http.StatusBadRequest, "Empty required field: action", w)
+	}
+	if inventoryTransaction.ItemId == "" {
+		return EncodeJSONStatus(http.StatusBadRequest, "Empty required field: item_id", w)
+	}
+	if inventoryTransaction.LocationId == "" {
+		return EncodeJSONStatus(http.StatusBadRequest, "Empty required field: location_id", w)
```
Comment (ssmall) [inline backend/api-service/src/api_inventory_service.go:184]: this could be a helper function given how many times it's needed in this file.

Comment (josueetcom) [reply backend/api-service/src/api_inventory_service.go]: Done.

File: backend/api-service/src/api_inventory_service.go [hunk 2]
```diff
@@ -251,3 +280,7 @@ func (s *InventoryApiService) UpdateLocation(id string, location Location, w htt
 
 	return EncodeJSONResponse(r, nil, w)
 }
+
+func encodeRequiredFieldMissing(name string, w http.ResponseWriter) error {
```
Comment (ssmall) [inline backend/api-service/src/api_inventory_service.go:284]: super-nit: `requiredFieldMissing` seems like a more idiomatic name

Comment (josueetcom) [reply backend/api-service/src/api_inventory_service.go]: Done

**Pull request merged**